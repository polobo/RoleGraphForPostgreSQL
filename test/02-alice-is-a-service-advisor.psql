-- In test 01 we introduced Alice and the Service Advisor job, here
-- we see what happens when we make Alice a member of the Service Advisor group.
-- Additionally, demonstrate the default behavior of createrole_self_grant
-- in granting set and inherit to admin by itself (since it has admin from superuser).

\if :is_cr
    SET createrole_self_grant TO set, inherit;
    SELECT format('CREATE ROLE u2_alice_%1$s LOGIN;', :'ctx') \gexec
    SELECT format('CREATE ROLE g2_service_advisor_%1$s NOLOGIN;', :'ctx') \gexec
    SELECT format('GRANT g2_service_advisor_%1$s TO u2_alice_%1$s;', :'ctx') \gexec
\endif

-- As shown in the visual graph there are now two paths that can be followed from Admin to Advisors
-- and this is reflected by the pair of service advisor memberof entries, one with a "via" and one
-- that is direct.
