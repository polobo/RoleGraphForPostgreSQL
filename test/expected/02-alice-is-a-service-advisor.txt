                                                                                   Test Result for 02-alice-is-a-service-advisor
        rolname        |             administration              |         memberof_users          |                   memberof_groups                    |              member_users               | member_groups 
-----------------------+-----------------------------------------+---------------------------------+------------------------------------------------------+-----------------------------------------+---------------
 g2_service_advisor_cr | by cr_admin from superuser              |                                 |                                                      | cr_admin via u2_alice_cr/cr_admin (s,i)+| 
                       |                                         |                                 |                                                      | cr_admin from cr_admin (s,i)           +| 
                       |                                         |                                 |                                                      | u2_alice_cr from cr_admin (s,i)         | 
 cr_admin              | Create Role                            +| u2_alice_cr from cr_admin (s,i) | g2_service_advisor_cr via u2_alice_cr/cr_admin (s,i)+|                                         | 
                       | of u2_alice_cr from superuser          +|                                 | g2_service_advisor_cr from cr_admin (s,i)            |                                         | 
                       | of g2_service_advisor_cr from superuser |                                 |                                                      |                                         | 
 u2_alice_cr           | by cr_admin from superuser              |                                 | g2_service_advisor_cr from cr_admin (s,i)            | cr_admin from cr_admin (s,i)            | 
(3 rows)

